pkg/model/bootstrapscript.go

```diff
@@ -61,31 +59,17 @@ func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup) (*fi.ResourceHo
 
 			return string(data), nil
 		},
-		// TODO temporary code, till vsphere cloud provider gets its own VFS implementation.
-		"Env1": func() string {
-			if b.AddAwsEnvironmentVariables && os.Getenv("AWS_REGION") != "" {
-				return "export AWS_REGION=" + os.Getenv("AWS_REGION")
-			}
-			return ""
-		},
-		"Env2": func() string {
-			if b.AddAwsEnvironmentVariables && os.Getenv("AWS_ACCESS_KEY_ID") != "" {
-				return "export AWS_ACCESS_KEY_ID=" + os.Getenv("AWS_ACCESS_KEY_ID")
-			}
-			return ""
-		},
-		"Env3": func() string {
-			if b.AddAwsEnvironmentVariables && os.Getenv("AWS_SECRET_ACCESS_KEY") != "" {
-				return "export AWS_SECRET_ACCESS_KEY=" + os.Getenv("AWS_SECRET_ACCESS_KEY")
+
+		// Pass in extra environment variables for user-defined S3 service
+		"S3Env": func() string {
+			if os.Getenv("S3_ENDPOINT") != "" {
+				return "export S3_ENDPOINT=" + os.Getenv("S3_ENDPOINT") + "\n" + "export AWS_REGION=" + os.Getenv("AWS_REGION") + "\n" + "export AWS_ACCESS_KEY_ID=" + os.Getenv("AWS_ACCESS_KEY_ID") + "\n" + "export AWS_SECRET_ACCESS_KEY=" + os.Getenv("AWS_SECRET_ACCESS_KEY")
 			}
 			return ""
 		},
 	}
 
 	nodeupTemplate := resources.AWSNodeUpTemplate
```

Comment:
Nit: nodeupTemplate is not really needed now, resources.AWSNodeUpTemplate can be passed directly.